T18936 include list of test-plan as sub-view for soc

alexandrasp opened the pull request:
Add a view showing all the test plans with the total number of test cases for
a particular SoC type.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
include list of test-plan as sub-view for soc

Add a view showing all the test plans
with the total number of test cases for
a particular SoC type.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

alexandrasp commented:
@gctucker I think that there is a problem in counts on this PR(test count), I am checking this, I will see how it works on staging. If you have any idea about the problem, let me know...

gctucker commented:
@alexandrasp There's a problem in the commit summary: `include list of test-plan assub-view`

gctucker commented:
@alexandrasp Also there's a Javascript error so it's failing to start the frontend service:
```
Error: Error: Parse error using esprima for file: /tmp/assets-build/js/app/view-socs-soc-job-kernel.2017.4.js
Error: Line 603: Unexpected token )
```

gctucker reviewed (changes requested):
Please fix Javascript error and clarify git commit summary.

Commit:
update count details for real results

update count details only if there is any
value as result and avoid showing alert
message on console in cases that have no
result.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

gctucker commented:
Tested on staging: https://staging.kernelci.org/soc/rockchip/job/kernelci/kernel/staging-20200213.0/

gctucker commented:
There are still several issues wit this view:
* The pie chart is showing a number of "boots" rather than "tests".
* The total numbers don't match what is expected, showing 5/0/0 instead of 468/5/442 on the example from the link above.
* This view should not have the list of expandable panels "Boards Tested" section but only the "Available Test Plans" to be like the Tests views.  The next view would show "Boards Tested" when selecting a single test plan.
* The numbers in the "Available Test Plans" don't seem correct, they should add up to 468/5/442.  It looks like `mach` is not used in the filters and all the results for that kernel are being counted.

gctucker reviewed (changes requested):
Please fix the issues as per my previous comment.

Commit:
update pie-chart from boot to test

update pie-chart used to show boot counts in soc view
to show a test count overview.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
remove boards sub-view and dead code

delete sub-view board tested from soc
views and all code related to this
feature.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
fix request to filter by mach

add soc/mach as filter in requests that will
get test groups and cases.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

alexandrasp commented:
> There are still several issues wit this view:
> 
>     * The pie chart is showing a number of "boots" rather than "tests".
> 
>     * The total numbers don't match what is expected, showing 5/0/0 instead of 468/5/442 on the example from the link above.
> 
>     * This view should not have the list of expandable panels "Boards Tested" section but only the "Available Test Plans" to be like the Tests views.  The next view would show "Boards Tested" when selecting a single test plan.
> 
>     * The numbers in the "Available Test Plans" don't seem correct, they should add up to 468/5/442.  It looks like `mach` is not used in the filters and all the results for that kernel are being counted.

I suppose all those were fixed ;)

alexandrasp commented:
@gctucker seems that is not in staging yet.

gctucker commented:
> @gctucker seems that is not in staging yet.

@alexandrasp It is now :)

alexandrasp commented:
@gctucker I detected a problem on pie-chart count, fixing it! Let me know if you find anything else..

Commit:
update page title

add soc as info in the title on
new view of soc test plans.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
show git url as link

update soc test plans view to show
git url as a clickable link.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

alexandrasp commented:
@gctucker could you update staging? Padovan suggested changing the title and make git url as link and I considered good points. What do you think?

gctucker commented on app/dashboard/static/css/bootstrap-3.3.6.min.css:
```diff
@@ -3,4 +3,4 @@
  * Copyright 2011-2015 Twitter, Inc.
  * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
  *//*! normalize.css v3.0.3 | MIT License | github.com/necolas/normalize.css */html{font-family:sans-serif;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%}body{margin:0}article,aside,details,figcaption,figure,footer,header,hgroup,main,menu,nav,section,summary{display:block}audio,canvas,progress,video{display:inline-block;vertical-align:baseline}audio:not([controls]){display:none;height:0} ...
```
Why is this change needed?

alexandrasp replied:
@gctucker 'Cause it was showing a warning message on console, this file was trying to get the map with a wrong name.

gctucker replied:
OK, that's a bit surprising as it has been used for a long time in other places.  I'll try to reproduce this while I'm working on the frontend.

gctucker replied:
@alexandrasp OK I guess this is because it was probably generated using a symlink originally.  Did you see the issue because you tried to regenerate it?  I mean, this is only a comment so it shouldn't cause any browser errors.

In any case, this is not related to the SoC view changes.  So please move this to a separate PR if it's really needed.

alexandrasp replied:
sure, already removed from this PR

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -20,381 +23,45 @@
+    'utils/table',
+    'tables/test',
     'utils/date',
     'sprintf'
```
These 2 are not used as far as I can tell, in which case they can be dropped from the list.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -20,381 +23,45 @@
-
     function getBootsFailed() {
         html.replaceContent(
             document.getElementById('accordion-container'),
```
I don't think there are any `accordion-container` elements any more in this view.  Can you locally get this function called instead of `getBootsDone` to make it show the error message?

alexandrasp replied:
This method is not necessary anymore, I removed it! Now the code is using same logic from job page, as you can see here: https://github.com/alexandrasp/kernelci-frontend/blob/t18936-add-soc-sub-view-test-plan/app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js#L429

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -20,381 +23,45 @@
-        }
-    }
-
     function getBootsFailed() {
```
It's not boots any more, we could call it `plans` instead i.e. `getPlansFailed`.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -20,381 +23,45 @@
-    }
-
     function updateDetails(response) {
+        results = response.result;
```
This variable is now set before it is defined.  I think the original line below should be left where it is.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -545,14 +204,6 @@ define([
-        });
-    }
-
     function getBoots() {
```
Rename to `getPlans`.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -591,83 +230,345 @@ define([
+            var mach = gSoc;
+            var qStr;
+
+            qStr = 'job=' + job;
```
Please use `URI.buildQuery` as in this example:
https://github.com/kernelci/kernelci-frontend/blob/master/app/dashboard/static/js/app/view-tests-job-branch-kernel-plan.2020.1.js#L362

It's a much safer way of constructing the query because it will encode special characters.  I hit an issue with the simple approach of creating a plain string when defconfig values had `+` and `=` characters in them.

alexandrasp replied:
updated.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -591,83 +230,345 @@ define([
+            var mach = gSoc;
+            var qStr;
+
+            qStr = 'job=' + job;
```
Same comment about `URI.buildQuery`.

alexandrasp replied:
also updated.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -545,14 +204,6 @@ define([
-
     function getBoots() {
         var deferred;
 
```
These boot results are not needed, and in fact they're not used anywhere.  You should just have to call `getPlans`.

gctucker reviewed (changes requested):
The test numbers look fine now on this view:

    https://staging.kernelci.org/soc/rockchip/job/kernelci/kernel/staging-20200213.0/

However some code about getting boot results needs to be removed, it seems to be a bit mixed up between the original file and the test results view.  There are also some minor things to clean-up.

gctucker commented:
@alexandrasp Staging is updated at least once a day, no need to request it each time.  It's not fully automated yet but it will be soon.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job.2017.3.3.js:
```diff
@@ -244,8 +249,6 @@ require([
         var columns,
             results;
-
-        console.log(response)
```
Erm, yes, sorry :angel:

gctucker reviewed (commented):
Please remove the "boot reports" links as these are going to be removed.
https://github.com/kernelci/kernelci-frontend/pull/70/files#diff-e4600ee66b5546cf4cba44d827b5be6aR100-R106
https://github.com/kernelci/kernelci-frontend/pull/70/files#diff-e4600ee66b5546cf4cba44d827b5be6aR133-R141

gctucker commented:
> Please remove the "boot reports" links as these are going to be removed.
> https://github.com/kernelci/kernelci-frontend/pull/70/files#diff-e4600ee66b5546cf4cba44d827b5be6aR100-R106
> https://github.com/kernelci/kernelci-frontend/pull/70/files#diff-e4600ee66b5546cf4cba44d827b5be6aR133-R141

I mean, the boot views are going to be removed, so we shouldn't be adding links to them now.

We may have some test views to link later on, but we can add them when we have them.

Commit:
remove boot report links

the boot views are going to be removed,
so we shouldn't have links to them now.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

alexandrasp commented:
> the boot views are going to be removed, so we shouldn't be adding links to them now.

I see and code is updated :)

gctucker commented:
Please make each row in the table link to a test view with the SoC name in the query, for example:
https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200221.0/plan/baseline/?qemu

This should then appear in the search box like it does with the jobs page:
https://kernelci.org/job/?mainline

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -20,380 +23,36 @@
-        }
-    }
-
     function updateDetails(response) {
```
I think the argument `response` should be renamed `results` to avoid ambiguity.  The results are taken from the response.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -402,19 +61,15 @@ define([
 
-        results = response.result;
-        if (results.length === 0) {
+        if (response.length === 0) {
```
This is still needed?  The function that calls `updateDetails` is already checking for the length of results and this variable is in fact `response.results[0]`.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -545,129 +169,340 @@ define([
+        var deferred;
+
+        function createBatchOp(result) {
+            
```
Superfluous blank line.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -545,129 +169,340 @@ define([
+            return
+        }
+    
+        updateDetails(response.result[0]);
```
See my comment below, about the value passed to `updateDetails`.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -701,10 +536,15 @@ define([
+        tableDivId: 'table-div',
+    });
+
+    getPlans();
```
Maybe add timeouts like on other views?  Is there any good reason to not have timeouts here?

gctucker reviewed (commented):
A few more minor things to tweak.

Commit:
add search filter to soc plan view

include the search filter on soc plan
view and put searched item in the search box

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
link to plan test cases

add a link to next view that will
be responsible to show all test
cases available in a plan.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -701,10 +533,15 @@ define([
+    setTimeout(getTestCount, 10);
 
     init.hotkeys();
     init.tooltip();
```
And, how about `setTimeout` for these functions?  They have timeouts in other views.

alexandrasp replied:
good point, updated.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -20,654 +23,483 @@
+            .data(results)
+            .columns(columns)
+            .order([0, 'asc'])
+            .rowURL('/soc/%(soc)s/job/%(job)s/kernel/%(kernel)s/plan/%(name)s/')
```
The `soc` value is undefined, causing URLs like this one:
```
https://staging.kernelci.org/soc/null/job/kernelci/kernel/staging-20200224.2/plan/baseline/
```
with `/soc/null/`.  Maybe use `gSoC`, or `mach` from the query data?

alexandrasp replied:
also updated.

gctucker commented on app/dashboard/static/js/app/view-socs-soc-job-kernel.2017.4.js:
```diff
@@ -20,654 +23,483 @@
+                'git_branch',
+                'git_commit',
+                'git_url',
+                'kernel',
```
Maybe add `mach` here to have it in `updatePlansTable()` and fix the URLs?

alexandrasp replied:
updated

gctucker reviewed (commented):
One more small thing about URLs from the table rows.

alexandrasp commented:
@gctucker is there any link to this new page?

gctucker reviewed (approved):
Tested OK on staging.

gctucker commented:
Example view on staging:
https://staging.kernelci.org/soc/rockchip/job/kernelci/kernel/staging-20200213.0/
